pkg/controller/config/match/match.go

```diff
@@ -0,0 +1,65 @@
+package match
+
+import (
+	configv1alpha1 "github.com/open-policy-agent/gatekeeper/apis/config/v1alpha1"
+)
+
+type Operation string
+
+const (
+	Audit   = Operation("audit")
+	Sync    = Operation("sync")
+	Webhook = Operation("webhook")
+	Star    = Operation("*")
+)
+
+type Set struct {
+	ExcludedNamespaces map[Operation][]string
+}
+
+var allOperations = []Operation{
+	Audit,
+	Webhook,
+	Sync,
+}
+
+var configMapSet = &Set{
+	ExcludedNamespaces: make(map[Operation][]string),
+}
+
+func NewSet() *Set {
+	return configMapSet
+}
+
+func (s *Set) Update(entry []configv1alpha1.MatchEntry) {
+	for _, matchEntry := range entry {
+		for _, ns := range matchEntry.ExcludedNamespaces {
+			for _, op := range matchEntry.Operations {
+				if Operation(op) == Star {
+					for _, o := range allOperations {
+						if !s.contains(o, ns) {
+							s.ExcludedNamespaces[o] = append(s.ExcludedNamespaces[o], ns)
+						}
+					}
+				} else {
+					if !s.contains(Operation(op), ns) {
+						s.ExcludedNamespaces[Operation(op)] = append(s.ExcludedNamespaces[Operation(op)], ns)
+					}
+				}
+			}
+		}
+	}
+}
+
+func (s *Set) Reset() {
```

Comment:
Reset() is dangerous to have externally callable, as it would cause all the controllers to see the match set as temporarily empty, causing them to enforce against all namespaces.

Reply:
this is replaced with `Replace()` that includes locks